ee/tabby-webserver/src/service/analytic.rs

```diff
@@ -70,6 +71,41 @@ impl AnalyticService for AnalyticServiceImpl {
             .collect();
         Ok(stats)
     }
+
+    async fn storage_stats(&self) -> Result<StorageStats> {
+        Ok(StorageStats {
+            events: recursive_dir_size(tabby_common::path::events_dir()).await?,
+            indexed_repositories: recursive_dir_size(tabby_common::path::dataset_dir())
+                .await?
+                .combine(recursive_dir_size(tabby_common::path::index_dir()).await?),
+            database: recursive_dir_size(crate::path::tabby_ee_root()).await?,
+            models: recursive_dir_size(tabby_common::path::models_dir()).await?,
+        })
+    }
+}
+
+/// Calculate the size of a directory in kilobytes recursively
+fn recursive_dir_size(
```

Comment:
consider use https://docs.rs/fs_extra/1.2.0/fs_extra/dir/fn.get_size.html

Reply:
Not async?

Reply:
use spawn blocking?